src/cmd/manage_nsfs.js

```diff
@@ -353,25 +344,64 @@ async function account_management(argv, config_root, from_file) {
     await manage_account_operations(action, data, config_root, config_root_backend);
 }
 
+/**
+ * set_access_keys will set the access keys either given as args or generated.
+ * @param {{ access_key: any; secret_key: any; }} argv
+ * @param {boolean} generate a flag for generating the access_keys automatically
+ */
+function set_access_keys(argv, generate) {
+    const { access_key, secret_key } = argv;
+    let generated_access_key;
+    let generated_secret_key;
+    if (generate) {
+        ({ access_key: generated_access_key, secret_key: generated_secret_key } = cloud_utils.generate_access_keys());
+        generated_access_key = generated_access_key.unwrap();
+        generated_secret_key = generated_secret_key.unwrap();
+    }
+
+    return [{
+        access_key: access_key || generated_access_key,
+        secret_key: secret_key || generated_secret_key,
+    }];
+}
+
+/**
+ * Will print the access_keys
+ * @param {{ access_keys: any[]; }} data
+ */
+function print_access_keys(data) {
+    const access_keys = data.access_keys[0];
+    console.log(`\n\n`);
+    console.log('access_key:', access_keys.access_key.unwrap());
+    console.log('secret_key:', access_keys.secret_key.unwrap());
+}
+
 async function fetch_account_data(argv, config_root, from_file) {
     let data;
+    let generate_access_keys = true;
     const action = argv._[1] || '';
     if (from_file) {
         const raw_data = await fs.promises.readFile(from_file);
         data = JSON.parse(raw_data.toString());
     }
+    let new_access_key = argv.new_access_key;
+    if (action === 'update') {
```

Comment:
can't we do this whole logic inside set_access_key()?

Reply:
We can, but that means that we need to get `set_access_key()` awareness for the action.
also it means that we will need to change `argv.new_access_key` or return `new_access_key` and we need to make sure we also return generate_access_keys as false due to the fact that if we do update then those keys should not be generated in the `access_keys` fields. 

I think it is more clear this way when the `set_access_key()` is not aware of the action.